docs/source/architecture/events_and_transactions_receipts.rst

```diff
@@ -0,0 +1,343 @@
+*******************************
+Events and Transaction Receipts
+*******************************
+
+Hyperledger Sawtooth supports the creating and broadcasting of Events.
+This allows application developers to do the following:
+
+- Subscribe to events that occur related to the blockchain, such
+  as a new block being committed or switching to a new fork.
+- Subscribe to application specific events defined by a transaction family.
+- Relay information about the execution of a transaction back to
+  clients without storing that data in state.
+
+It is important to note the following constraints:
+
+- Events are transmitted only on block boundaries.
+- Event notifications are implemented internally using the Observer pattern.
+- Both internal and external event subscribers are supported.
+
+.. image:: ../images/event_subsystem.*
+   :width: 80%
+   :align: center
+   :alt: Event Subsystem
+
+Events
+======
+
+Events will be represented with the following protobuf message:
+
+.. code-block:: protobuf
+
+  message Event {
+    // Used to subscribe to events and servers as a hint for how to deserialize
+    // event_data and what pairs to expect in attributes.
+    string event_type = 1;
+
+    // Transparent data defined by the event_type.
+    message Attribute {
+      string key = 1;
+      string value = 2;
+    }
+    repeated Attribute attributes = 2;
+
+    // Opaque data defined by the event_type.
+    bytes  data = 3;
+  }
+
+
+Events are extracted from other data structures such as blocks or
+transaction receipts. In order to treat this extraction uniformly, an
+EventExtractor interface is used. This interface is implemented for
+each event source. The EventExtractor interface takes a list of
+EventSubscriptions and will only generate events that are in the union of all
+subscriptions. An event is "in a subscription" if the event's event_type field
+matches the subscription's event_type field and, if the subscription has any
+filters, the key-value pair of the filter matches the key-value pair of one of
+the key-value pairs in the event's meta_data field. EventExtractor
+implementations should be lightweight and reusable across the system.
```

Comment:
I'm confused by this last sentence: "EventExtractor implementations should be lightweight and reusable across the system."  Is this guidance for the application developer?  If so, perhaps more words would help provide a segue.  Or perhaps it was an implementation goal from the design doc?  In that case, I'd remove this sentence.

Reply:
remove